File: shared/src/main/java/com/batu/shared/dto/request/AccountRequestDto.java
package com.batu.shared.dto.request;

import java.math.BigDecimal;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AccountRequestDto {

    private UUID accountId;
    private UUID connectionId;
    private UUID userId;
    private String institutionName;
    private String accountName;
    private String accountType;
    private String accountSubtype;
    private String accountMask;
    private BigDecimal currentBalance;
    private BigDecimal availableBalance;
    private String isoCurrencyCode;
    @JsonProperty("isActive")
    private boolean isActive = true;
}

File: shared/src/main/kotlin/com/batu/shared/dto/AccountRequestDto.kt
package com.batu.shared.dto

import java.util.UUID
import java.time.LocalDateTime
import java.math.BigDecimal


data class AccountRequestDto(
    val connectionId : UUID,
    val userId : UUID,
    val externalId : String,
    val accountName : String,
    val accountType : String,
    val accountSubtype : String,
    val accountMask : String,
    val currentBalance : BigDecimal,
    val availableBalance : BigDecimal,
    val isoCurrencyCode : String,
    val isActive : Boolean? = true
){

}